File: reactjs-app/src/services/NotificationService.js
import httpAxios from '../httpAxios';

const NotificationService = {
    create: (notification) => {
        return httpAxios.post(`user-services/api/notifications/create`, notification);
    },
    getById: (id) => {
        return httpAxios.get(`user-services/api/notifications/get-by-id/${id}`);
    },
    getByUser: (id) => {
        return httpAxios.get(`user-services/api/notifications/get-by-user/${id}`);
    },
    getAll: () => {
        return httpAxios.get(`user-services/api/notifications/get-all`);
    },
    countUnseen: (id) => {
        return httpAxios.get(`user-services/api/notifications/count-unseen-by-user/${id}`);
    },
    update: (id, notification) => {
        return httpAxios.put(`user-services/api/notifications/update/${id}`, notification);
    },
    trash: (id) => {
        return httpAxios.put(`user-services/api/notifications/trash/${id}`);
    },
    delete: (id) => {
        return httpAxios.delete(`user-services/api/notifications/delete/${id}`);
    },
    switchStatus: (id) => {
        return httpAxios.put(`user-services/api/notifications/switch-status/${id}`);
    },
    seen: (id) => {
        return httpAxios.put(`user-services/api/notifications/seen/${id}`);
    },
};

export default NotificationService;

File: user-services/src/main/java/com/microservices/userservices/service/NotificationService.java
package com.microservices.userservices.service;

import java.util.List;
import java.util.UUID;

import com.microservices.userservices.payload.request.NotificationRequest;
import com.microservices.userservices.payload.response.NotificationResponse;

public interface NotificationService {

    NotificationResponse create(NotificationRequest notificationRequest);

    NotificationResponse getById(UUID id);

    List<NotificationResponse> getAll();
    
    NotificationResponse update(UUID id, NotificationRequest notificationRequest);

    NotificationResponse delete(UUID id);

    List<NotificationResponse> getByUser(UUID userId);

    void switchStatus(UUID id);

    void trash(UUID id);
}
